### PR TITLE
[Vengeance] add T30 2pc base implementation

### PR DESCRIPTION
**Pull request by ToppleTheNun**
**With current APL and T30 2pc (no fire damage amp applied)**
image

**With current APL and T30 2pc (fire damage amp applied)**
image

### CODE REVIEW

**Comment by ToppleTheNun** [comment]
@EvanMichaels

**Comment by EvanMichaels** [comment]
Heya, do we not have spell data yet for this? Is it just not implemented in-game? Or do we need to whitelist some things for the generator?

**Comment by ToppleTheNun** [comment]
VDH T30 isn't implemented in-game yet, I'm basing the implementation off of the announced tier bonus:
> (2) Set Bonus: Soul Fragments heal for 10% more and generating a Soul Fragment increases your Fire damage by 2% for 6 seconds. Multiple applications may overlap.
(4) Set Bonus: Shear and Fracture deal Fire damage. After consuming 20 Soul Fragments, your next cast of Shear or Fracture applies Fiery Brand to its target for 6 seconds.

**Comment by ToppleTheNun** [comment]
@EvanMichaels Implemented using the actual spell data.

**Before**
image

**After**
image

File: engine/class_modules/sc_demon_hunter.cpp [hunk 1]
```diff
@@ -6939,6 +6954,12 @@ double demon_hunter_t::composite_player_multiplier( school_e school ) const
 {
   double m = player_t::composite_player_multiplier( school );
 
+  if ( set_bonuses.t30_vengeance_2pc->ok() &&
+       set_bonuses.t30_vengeance_2pc_buff->effectN( 1 ).has_common_school( school ) )
+  {
+    m *= 1.0 + buff.t30_vengeance_2pc->check() * set_bonuses.t30_vengeance_2pc_buff->effectN( 1 ).percent();
```
Comment (EvanMichaels) [inline engine/class_modules/sc_demon_hunter.cpp]: I think you probably could simplify to `m *= 1.0 + buff.t30_vengeance_2pc->stack_value()`? Think you are loading the effect into the default value of the buff via the ->set_default_value_from_effect( 1 )?

File: engine/class_modules/sc_demon_hunter.cpp [hunk 2]
```diff
@@ -5761,11 +5767,18 @@ void demon_hunter_t::create_buffs()
...
           ->set_default_value_from_effect_type( A_MOD_TOTAL_STAT_PERCENTAGE )
           ->set_pct_buff_type( STAT_PCT_BUFF_AGILITY );
-  
+
   buff.t30_havoc_4pc =
       make_buff<buff_t>( this, "seething_potential",
                          set_bonuses.t30_havoc_4pc->ok() ? set_bonuses.t30_havoc_4pc_buff : spell_data_t::not_found() )
           ->set_default_value_from_effect( 1 );
+
+  buff.t30_vengeance_2pc = make_buff<buff_t>(
```
Comment (EvanMichaels) [inline engine/class_modules/sc_demon_hunter.cpp]: Given the description "Multiple applications may overlap." is this a normal stacking buff or should this be `->set_stack_behavior( buff_stack_behavior::ASYNCHRONOUS )`? Usually they only use this wording for async behavior.

File: engine/class_modules/sc_demon_hunter.cpp [hunk 3]
```diff
@@ -5761,11 +5767,18 @@ void demon_hunter_t::create_buffs()
...
+      this, "fires_of_fel", set_bonuses.t30_vengeance_2pc->ok() ? set_bonuses.t30_vengeance_2pc_buff : spell_data_t::not_found() )
+      ->set_default_value_from_effect( 1 )
+      ->add_invalidate( CACHE_PLAYER_DAMAGE_MULTIPLIER );
+  buff.t30_vengeance_4pc = make_buff<buff_t>(
```
Comment (EvanMichaels) [inline engine/class_modules/sc_demon_hunter.cpp:5783]: As a note, the set bonus passive "#3 (id=1074773) : Apply Aura (6) | Modify Spell School (220)" should be able to be done with `apply_affecting_aura( set_bonuses.t30_vengeance_4pc )` in the normal passive application spot to do the school modification.

**Review by EvanMichaels — CHANGES_REQUESTED** [review]

**Pull request merged**